File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0) return false; // negative numbers are not palindrome
        
        long rev = 0, temp = x;
        
        while (temp != 0) {
            rev = rev * 10 + temp % 10;
            temp /= 10;
        }
        
        return rev == x;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0){
            return false;// agar negative toh return false
        }

        int n = x; //x ko n me store kiya
        int revnum = 0;

        while(n>0){
            int d = n%10;
            revnum = revnum*10 +d;
            n = n/10;
        }

        if(revnum == x){
            return true;
        }else {
            return false;
        }
    
    }
}